### PR TITLE
[M3] Frequently Asked Questions

### LINKED ISSUE #40: [M3] [FAQ] Calculate most frequently asked search questions
# Overview
As part of this milestone, we'd like to add a frequently asked questions box to the help desk landing page. 

This task involves the backend calculation behind which queries are frequently asked. A few points of consideration include preventing the ability to _spam_ a query to get it appear on the FAQ list, as well as preventing nonsensical queries from appearing on the FAQ list. The initial implementation will have one global FAQ list that ignores which category is selected. The final goal is to have the FAQ list updated as a user selects a category.

### LINKED ISSUE #45: [M3] [FAQ] UI for frequently asked questions

### PR DESCRIPTION
**Pull request by najclark**
Creates a frequently asked questions feature where users can see what resources others have been viewing for a particular category, or for no category.

The FAQ is visible on the main search page. It updates to reflect what category the user selects when the user changes the category in the category dropdown. Each link can be clicked to go directly to the query.

image

Closes #40, Closes #45

### CODE REVIEW

File: src/helpdesk_proj/static/helpdesk_app/css/search.css
Comment (ayoubmansar) [inline src/helpdesk_proj/static/helpdesk_app/css/search.css:211]: As part of #78 efforts, can we change this to `width: 80%;` so that it looks cleaner on mobile?

File: src/templates/search.html
Comment (ayoubmansar) [inline src/templates/search.html:199]: This looks a bit large on mobile devices — could we do `h3` instead?
And perhaps add a rule for
```
.faq-box h3 {
    margin: 13px;
}
```
so the top doesn't look weird.

**Review by ayoubmansar — APPROVED** [review]
Looks good to me, left some small feedback about adjusting some of the sizing to be compatible with mobile devices. Going to approve this as to unblock you whenever you wish to merge.

File: src/helpdesk_proj/static/helpdesk_app/css/search.css
Comment (najclark) [reply src/helpdesk_proj/static/helpdesk_app/css/search.css:211]: For me it looks reasonable on mobile. With `width: 80%` I think it looks a little weird on desktop.

Current mobile:
image

Updated desktop:
image

I can revisit this after this PR and #78 is merged.

File: src/templates/search.html
Comment (najclark) [reply src/templates/search.html:199]: Changed to h3

**Review by teddyliang — APPROVED** [review]
lgtm

**Pull request merged**
